test/parallel/test-fs-watchfile.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const assert = require('assert');
+const common = require('../common');
+const fixtures = path.join(__dirname, '..', 'fixtures');
+
+function noop() {}
+
+var callbackFired = 0;
+
+// Basic usage tests.
+assert.throws(function() {
+    fs.watchFile('./some-file');
+}, /watchFile requires a listener function/);
+
+assert.throws(function() {
+    fs.watchFile('./another-file', {}, 'bad listener');
+}, /watchFile requires a listener function/);
+
+assert.throws(function() {
+    fs.watchFile(new Object(), function() {})
+}, /path must be a string/i);
+
+// Test ENOENT. Should fire once.
+var enoentFile = path.join(fixtures, 'empty', 'non-existent-file');
+fs.watchFile(enoentFile, function(curr, prev) {
+    callbackFired++;
+});
+
+setTimeout(function() {
+    fs.unwatchFile(enoentFile);
+}, common.platformTimeout(2));
```

Comment:
It was originally to prevent multiple callbacks, see [here](https://github.com/brendanashworth/io.js/commit/4d5655d73270ee480ad481e588404c87733384c0#commitcomment-11971872), but I'm changing it back to not wait.